### PR TITLE
refactor: simplify logic form control

### PR DESCRIPTION
**Pull request by dominik-stumpf**

### CODE REVIEW

File: src/components/create-guild/Requirements/components/LogicFormControl.tsx [hunk 1]
```diff
@@ -44,114 +21,54 @@ const LogicFormControl = ({
...
-
-            <FormErrorMessage>
-              {errors?.anyOfNum?.message?.toString()}
-            </FormErrorMessage>
-          </FormControl>
-        </HStack>
-      )}
+      <HStack alignItems="start" pt="px">
+        <Text as="span" flexShrink={0} pt={2}>
+          {pluralize(requirementCount, "requirement", false)} out of{" "}
```
Comment (BrickheadJohnny) [inline src/components/create-guild/Requirements/components/LogicFormControl.tsx:68]: I think we should pluralize `anyOfNum` here ;)

**Review by BrickheadJohnny — APPROVED** [review]
Wonderful cleanup, I only left 1 minor comment!

**Review by TomiOhl — COMMENTED** [review]
Nit: probably "All" should be treated as a separate case, "Should meet All requirement out of n" sounds a bit off -> it should either just cause `requirement` to be pluralized or maybe it should be "Should meet all requirements"?

**Comment by BrickheadJohnny** [comment]
> Nit: probably "All" should be treated as a separate case, "Should meet All requirement out of n" sounds a bit off -> it should either just cause `requirement` to be pluralized or maybe it should be "Should meet all requirements"?

Good idea, @dominik-stumpf could you please look into it?

**Comment by dominik-stumpf** [comment]
> Nit: probably "All" should be treated as a separate case, "Should meet All requirement out of n" sounds a bit off -> it should either just cause `requirement` to be pluralized or maybe it should be "Should meet all requirements"?

So it should be like this? @TomiOhl 

- "Should meet all requirements"
- "Should meet 3 requirements"
- "Should meet 1 requirement"

**Comment by TomiOhl** [comment]
> > Nit: probably "All" should be treated as a separate case, "Should meet All requirement out of n" sounds a bit off -> it should either just cause `requirement` to be pluralized or maybe it should be "Should meet all requirements"?
> 
> So it should be like this? @TomiOhl
> 
> * "Should meet all requirements"
> * "Should meet 3 requirements"
> * "Should meet 1 requirement"

Yes either this or:
* "Should meet all requirements"
* "Should meet 3 requirements out of n"
* "Should meet 1 requirement out of n"

**Pull request merged**